src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,141 @@
+     * be parsed into LocalDate and LocalTime objects if successful.
+     * @param description the description of the task.
+     * @param by the raw date time string of the task.
+     * @throws DukeException when the date time format is incorrect.
+     */
+    public Deadline(String description, String by) throws DukeException {
+        super("D", description);
+        try {
+            String[] dateTime = by.split(" ");
+            this.date = parseDate(dateTime[0]);
+            if (dateTime.length < 2) {
+                this.time = Optional.empty();
+            } else {
+                this.time = Optional.of(parseTime(dateTime[1]));
+            }
+        } catch (DateTimeParseException de) {
+            throw new DukeException("Invalid date or time format!");
+        } catch (ArrayIndexOutOfBoundsException aioobe) {
+            throw new DukeException("Missing date!");
+        }
+    }
+
+    /**
+     * Overloaded constructor for Deadline.
+     * This constructor is usually used when loading tasks from the txt file.
+     * @param description the description of the task.
+     * @param by the raw date time string of the task.
+     * @param isDone the completion flag of the task.
+     * @throws DukeException when the date time format is incorrect.
+     */
+    public Deadline(String description, String by, boolean isDone) throws DukeException {
+        super("D", description, isDone);
+        try {
+            String[] dateTime = by.split(" ");
+            this.date = parseDate(dateTime[0]);
+            if (dateTime.length < 2) {
+                this.time = Optional.empty();
+            } else {
+                this.time = Optional.of(parseTime(dateTime[1]));
+            }
+        } catch (DateTimeParseException de) {
+            throw new DukeException("Invalid date or time format!");
+        } catch (ArrayIndexOutOfBoundsException aioobe) {
+            throw new DukeException("Missing date!");
+        }
+    }
+
+    /**
+     * Helper method to parse raw date input into LocalDate. The method
+     * recognizes special keywords like "today" and "tomorrow" to improve
+     * user-friendliness.
+     * @param raw the raw date input.
+     * @return the LocalDate object representing the date.
+     */
+    private LocalDate parseDate(String raw) {
+        if (raw.equals("today")) {
+            return LocalDate.now();
+        } else if (raw.equals("tomorrow")) {
+            return LocalDate.now().plusDays(1);
+        }
+        return LocalDate.parse(raw);
+    }
+
+    /**
+     * Helper method to parse raw time input into LocalTime. The method
+     * can provide left padding of zero, if required, and accepts both
+     * ":" and "." as separators of hours and minutes.
+     * @param raw the raw input time.
+     * @return the LocalTime object representing the time.
+     */
+    private LocalTime parseTime(String raw) {
+        String t = String.format("%1$" + 5 + "s",
```

Comment:
Styling error. Consider using time as the name for the variable instead of t. It will communicate the meaning in an easier way.